easybuild/easyblocks/generic/intelbase.py

```diff
@@ -260,31 +269,42 @@ def install_step(self, silent_cfg_names_map=None, silent_cfg_extras=None):
         if silent_cfg_names_map is None:
             silent_cfg_names_map = {}
 
-        # license file entry is only applicable with license file or server type of activation
-        # also check whether specified activation type makes sense
-        lic_file_server_activations = [ACTIVATION_LIC_FILE, ACTIVATION_LIC_SERVER]
-        other_activations = [act for act in ACTIVATION_TYPES if act not in lic_file_server_activations]
-        lic_file_entry = ""
-        if self.cfg['license_activation'] in lic_file_server_activations:
-            lic_file_entry = "%(license_file_name)s=%(license_file)s"
-        elif not self.cfg['license_activation'] in other_activations:
-            raise EasyBuildError("Unknown type of activation specified: %s (known :%s)",
-                                 self.cfg['license_activation'], ACTIVATION_TYPES)
-
-        silent = '\n'.join([
-            "%(activation_name)s=%(activation)s",
-            lic_file_entry,
+        if self.requires_runtime_license:
+            # license file entry is only applicable with license file or server type of activation
+            # also check whether specified activation type makes sense
+            lic_file_server_activations = [ACTIVATION_LIC_FILE, ACTIVATION_LIC_SERVER]
+            other_activations = [act for act in ACTIVATION_TYPES if act not in lic_file_server_activations]
+            lic_file_entry = ""
+            if self.cfg['license_activation'] in lic_file_server_activations:
+                lic_file_entry = "%(license_file_name)s=%(license_file)s"
+            elif not self.cfg['license_activation'] in other_activations:
+                raise EasyBuildError("Unknown type of activation specified: %s (known :%s)",
+                                     self.cfg['license_activation'], ACTIVATION_TYPES)
+            silent = '\n'.join([
+                "%(activation_name)s=%(activation)s",
+                lic_file_entry,
+                ""  # Add a newline at the end, so we can easily append if needed
+            ]) % {
+                'activation_name': silent_cfg_names_map.get('activation_name', ACTIVATION_NAME),
+                'license_file_name': silent_cfg_names_map.get('license_file_name', LICENSE_FILE_NAME),
+                'activation': self.cfg['license_activation'],
+                'license_file': self.license_file,
+            }
+        else:
+            self.log.debug(
+                "Ignoring license checks: self.requires_runtime_license=%s self.cfg['requires_runtime_license']=%s"
+                % (self.requires_runtime_license, self.cfg['requires_runtime_license'])
+            )
+            silent = '\n'
+
+        silent = silent.join([
```

Comment:
this doesn't look right, this should be:

```
silent += '\n'.join([
```

Otherwise you'll be repeating the text that specifies the license several times.

See:

```python
>>> 'x'.join(['a', 'b', 'c'])
'axbxc'
```